# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: c2741585119b
# Full Hash: c2741585119ba8536918b94429296479c0865c25
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-07-27 09:45:43
# Regressor Bug: 1695806
# File Overlap Count: 1
# Description:
#   Bug 1695806 - Defer sending a snapshot reqeust for remote process iframe until cloning the iframe document finished. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119950
# ==============================================================================

diff -r 3c5186df3677 -r c2741585119b docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Tue Jul 27 00:47:44 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Tue Jul 27 00:47:44 2021 +0000
@@ -36,6 +36,10 @@
 namespace mozilla {
 enum class CallState;
 
+namespace embedding {
+class PrintData;
+}
+
 namespace net {
 class DocumentLoadListener;
 }
@@ -331,6 +335,16 @@
   void ClearPermanentKey() { mPermanentKey.setNull(); }
   void MaybeSetPermanentKey(Element* aEmbedder);
 
+  void CloneDocumentTreeInto(CanonicalBrowsingContext* aSource,
+                             const nsACString& aRemoteType,
+                             embedding::PrintData&& aPrintData);
+
+  // Returns a Promise which resolves when cloning documents for printing
+  // finished if this browsing context is cloning document tree.
+  RefPtr<GenericNonExclusivePromise> GetClonePromise() const {
+    return mClonePromise;
+  }
+
  protected:
   // Called when the browsing context is being discarded.
   void CanonicalDiscard();
@@ -494,6 +508,9 @@
 
   bool mIsReplaced = false;
 
+  // A Promise created when cloning documents for printing.
+  RefPtr<GenericNonExclusivePromise> mClonePromise;
+
   JS::Heap<JS::Value> mPermanentKey;
 };
 